examples/join/weighted.ts

```diff
@@ -64,12 +63,12 @@ const join = async () => {
     );
 
     const poolJoin = new PoolJoin();
-    const poolTokens = poolState.tokens.map(
-        (t) => new Token(chainId, t.address, t.decimals),
-    );
-    const amountsIn = poolTokens.map((t) =>
-        TokenAmount.fromHumanAmount(t, '1'),
-    );
+
+    const amountsIn = poolState.tokens.map((t) => ({
+        rawAmount: parseUnits('1', t.decimals),
+        decimals: t.decimals,
+        address: t.address,
```

Comment:
I personally like the `TokenAmount` [value object](https://martinfowler.com/bliki/ValueObject.html) abstraction and I think that using it (or a similar abstraction) in the internals of the frontend could be useful. (if you already have a token instance, having methods such as `fromHumanAmount` make the code easier to read).

That said, from an API consumer point of view, I think that it is better to leave more flexibility and freedom to the user and I agree that allowing primitives is better.

Reply:
It might be useful to add a `toInputAmount` to the TokenAmount?

Reply:
I think it could be handy. Maybe also for you to use it in the sdk internals? 🤔